### PR TITLE
[DO-NOT-MERGE] UM/UP for Hub - combined remaining changes [DPP-1147][DPP-1198]

### PR DESCRIPTION
**Pull request by pbatko-da**
LAPI design only:   https://github.com/digital-asset/daml/pull/14550

### CODE REVIEW

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala [hunk 1]
```diff
@@ -47,6 +47,11 @@ final class UserManagementServiceImpl extends UserManagementService with FakeAut
 
   override def listUserRights(request: ListUserRightsRequest): Future[ListUserRightsResponse] =
     record(request)(ListUserRightsResponse.defaultInstance)
+
+  override def updateUser(request: UpdateUserRequest): Future[UpdateUserResponse] = {
+    // TODO: um-for-hub
+    ???
```
Comment (mziolekda) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala:53]: These are mocks, so a trivial impl needs to be provided

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  // When used in update API calls with a field mask can only select the whole 'annotations' map with the replace-all semantics.
+  // I.e. one cannot select a particular key within the 'annotations' map.
+  //
+  // Duplicate keys are disallowed by the semantics of the protobuf3 maps.
+  // See: https://developers.google.com/protocol-buffers/docs/proto3#maps
+  //
+  // Optional
+  map<string, string> annotations = 12;
+
+}
```
Comment (mziolekda) [inline ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto:53]: newline

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala (hunk 1 above)
Comment (pbatko-da) [reply language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/grpc/helpers/UserManagementServiceImpl.scala]: Ok

File: ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto (hunk 2 above)
Comment (pbatko-da) [reply ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto:62]: fixed

**Pull request closed without merging**